src/utils/utils.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <QtGlobal>
+
+namespace Utils {
+/**
+ * @brief Operating system type.
+ * Always test for a match using '&' instead of '=='.
+ * @example
+ * ```
+ * Utils::Os os = Utils::MACOS;
+ * bool isUnix = os & Utils::UNIX;  // true (correct usage)
+ * bool isUnix = os == Utils::UNIX; // false (incorrect usage)
+ * ```
+ */
+enum Os
```

Comment:
I generally like using a constexpr value for this instead of #ifdefs scattered everywhere. However, I think the API should change to use ==. Testing with & does not feel natural.

Reply:
@borgmanJeremy On the other hand it allows us to test for `os & UNIX`.
If `os` is `MACOS` then `os == UNIX` will be false.

Reply:
@veracioux something like this maybe? https://godbolt.org/z/Y4n39qr48

Reply:
@borgmanJeremy I abandoned this idea altogether. It didn't make sense in those places where system-specific includes were needed. With those cases taken out of the picture, the few that were left don't warrant the trouble. Btw, this PR is ready to merge as far as I'm concerned.